crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -975,28 +975,45 @@ impl SlidingSync {
         self.rooms.read().unwrap().values().cloned().collect()
     }
 
+    /// Handle the HTTP response.
+    ///
+    /// But which response? `v4::Response`, aka the Sliding Sync response, or
+    /// `SyncResponse` which still relies on `v3`?
+    /// Well that's tricky. We have both here, because this Sliding Sync
+    /// implementation is still experimental, and we didn't want to be too
+    /// invasive. Thus, `SyncResponse` doesn't support Sliding Sync yet. Hence
+    /// the fact this method handles both at the same time. It's not super
+    /// annoying but it was important to clarify that.
```

Comment:
I remember we have some code path that's common between the two sync types, but this fn is in sliding_sync so it can't be used from elsewhere. This comment adds confusion when it should be clarifying things 😅

Reply:
Updated.